core/core.go

```diff
@@ -597,22 +598,27 @@ func (e ErrSnapshotNotFound) Error() string {
 	return fmt.Sprintf("No snapshot found with id: %s. To create a new snapshot, send a request with an empty snapshotID", e.id)
 }
 
+// ErrPerPageZero is the error returned when a GetOrders request specifies perPage to 0
+type ErrPerPageZero struct{}
+
+func (e ErrPerPageZero) Error() string {
```

Comment:
I don't think we need a special type for this error. In general, you only need to use typed errors for cases where the caller might be expected to programmatically react to the error. In other words: "If the function returns this error, I need to do this, but if it returns a different error I need to do that." I can't imagine that being the case here.

If you _do_ want to still use a specific named error, this should just be a variable. Error types should be used when you need to include additional information about the error, e.g., in struct fields. Otherwise, you can just use a variable. See [this guide on naming errors](https://github.com/golang/go/wiki/Errors#naming). I would also suggest tweaking the language a bit since it is also returned when `perPage` is less than zero.

```
var ErrInvalidPerPage = errors.New("perPage cannot be zero or less")
```

Though again, IMO it makes more sense to just return `errors.New("perPage cannot be zero or less")` inside the function instead of declaring this ahead of time.

Reply:
We do programmatically react to this error. See line 88 in `rpc_handler.go`

Reply:
Ah sorry I missed that. Makes sense to me.